framework_crates/bones_framework/src/render/audio.rs

```diff
@@ -1,29 +1,364 @@
 //! Audio components.
 
+use std::collections::VecDeque;
 use std::io::Cursor;
+use std::time::Duration;
 
 use crate::prelude::*;
-
-pub use kira::{self, sound::static_sound::StaticSoundData};
+use kira;
+pub use kira::sound::static_sound::StaticSoundData;
 use kira::{
     manager::{
         backend::{cpal::CpalBackend, mock::MockBackend, Backend},
         AudioManager as KiraAudioManager,
     },
-    sound::SoundData,
+    sound::{
+        static_sound::{StaticSoundHandle, StaticSoundSettings},
+        PlaybackState, SoundData,
+    },
+    tween,
+    tween::Tween,
+    Volume,
 };
+use tracing::warn;
 
-/// The game plugin for the audio system.
+/// The amount of time to spend fading the music in and out.
+pub const MUSIC_FADE_DURATION: Duration = Duration::from_millis(500);
```

Comment:
Do you think maybe we should make this configurable somehow?

Reply:
Yeah we can throw it into AudioCenter with a getter/setter + default.